uk_tax_23-24.py

```diff
@@ -80,16 +83,27 @@ def national_insurance(incomes):
 
     return ni
 
-def student_loan_repayment_plan_2(incomes):
-    repayments = np.where(
-        incomes > student_loan_plan_2_threshold,
-        (incomes - student_loan_plan_2_threshold) * student_loan_plan_2_rate,
-        0,
-    )
-
+# Can expand for further student loan repayment plans (3/4/5) in future
+def student_loan_repayment_plan(incomes, student_loan_plan_number=2):
+    if student_loan_plan_number == 1:
+        repayments = np.where(
+            incomes > student_loan_plan_1_threshold,
+            (incomes - student_loan_plan_1_threshold) * student_loan_plan_1_rate,
+            0,
+        )
+    elif student_loan_plan_number == 2:
+        repayments = np.where(
+            incomes > student_loan_plan_2_threshold,
+            (incomes - student_loan_plan_2_threshold) * student_loan_plan_2_rate,
+            0,
+        )
+    else:
+        print(f"We don't account for Student Loan plan number {student_loan_plan_number} yet - repayments will be 0")
+        repayments = 0
+    print(f"Student loan plan number is {student_loan_plan_number} and total repayments are \033[1;34m£{repayments}\033[0m")
     return repayments
 
-def calculate_taxes(incomes, pension_contrib_percent=0, voluntary_pension_contrib=0, include_student_loan=False):
+def calculate_taxes(incomes, pension_contrib_percent=0, student_loan_plan_number=2, voluntary_pension_contrib=0, include_student_loan=False):
```

Comment:
I would prefer if the Repayment Plan is a string ("Plan 2", "Postgraduate"), as it is more verbose and gives better support for the Postgrad Loan Repayments.